network/p2p/peerID.go

```diff
@@ -0,0 +1,94 @@
+// Copyright (C) 2019-2023 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+// Package node is the Algorand node itself, with functions exposed to the frontend
+
+package p2p
+
+import (
+	"crypto/rand"
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/libp2p/go-libp2p/core/crypto"
+
+	"github.com/algorand/go-algorand/config"
+)
+
+// DefaultPrivKeyPath is the default path inside the node's root directory at which the private key
+// for p2p identity is found and persisted to when a new one is generated.
+const DefaultPrivKeyPath = "peerIDPrivKey.pem"
+
+// GetPrivKey manages loading and creation of private keys for network PeerIDs
+// It prioritizes, in this order:
+//  1. user supplied path to privKey
+//  2. default path to privKey,
+//  3. generating a new privKey.
+//
+// If a new privKey is generated it will be saved to default path if cfg.P2PPersistPeerID.
+func GetPrivKey(cfg config.Local, dataDir string) (crypto.PrivKey, error) {
```

Comment:
nit: do we want to store and return peer.ID?

Reply:
I left it as PrivKey since when you construct a libp2p host [here](https://github.com/libp2p/go-libp2p/blob/master/config/config.go#L84C1-L85C1) you pass it the PrivKey and it [gets the peerID from that](https://github.com/libp2p/go-libp2p/blob/master/config/config.go#L153) and stores it in the peerstore.